test/install_test.go

```diff
@@ -131,7 +131,7 @@ func TestVersionPostInstall(t *testing.T) {
 }
 
 func TestCheckPostInstall(t *testing.T) {
-	cmd := []string{"check", "--expected-version", TestHelper.GetVersion(), "--wait=0"}
+	cmd := []string{"check", "--expected-version", TestHelper.GetVersion(), "--wait=1m"}
```

Comment:
We were previously using `--wait=0` here, since the check output would be different if any of the checks had to retry, and the subsequent golden file check would fail. It looks like that's still the case. If I run `linkerd check` while the control plane is installing and dump the output to a text file, that file contains a bunch of control characters that are printed when checks are retried:

```
linkerd-existence
-----------------
√ control plane namespace exists
| controller pod is running -- waiting for check to complete [K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K/ controller pod is running -- waiting for check to complete [K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K- controller pod is running -- waiting for check to complete 
```

Am pretty sure those control characters will interfere with the `ValidateOutput` call on line 146. Would you mind reverting the changes to the `--wait` flag in files file, and instead wrapping the those tests with `TestHelper.RetryFor` to achieve the retry behavior that you want?

Reply:
Good catch @klingerf. I optimistically assumed the output would ignore the spinners. Will revert.